### PR TITLE
Fix delete children on removed element

### PR DESCRIPTION
**Pull request by viktorianer**
VN-1879

### CODE REVIEW

**Comment by viktorianer** [comment]
Added feature toggle for Modeling-Templates. Please review the changes.
In order to add this toggle to vn-1886 please merge this in develop branche.

File: sernet.gs.service/src/sernet/verinice/model/common/CnATreeElement.java [hunk 1]
```diff
@@ -268,8 +268,15 @@ public void remove() {
...
-		CopyOnWriteArrayList<CnALink> list2 = new CopyOnWriteArrayList<CnALink>(getLinksDown());
+		removeCnLinks();
+	}
+
+    /**
+     * Remove all links from and to this item.
+     * 
+     */
+    public void removeCnLinks() {
+        CopyOnWriteArrayList<CnALink> list2 = new CopyOnWriteArrayList<CnALink>(getLinksDown());
```
Comment (neosimsim) [inline sernet.gs.service/src/sernet/verinice/model/common/CnATreeElement.java]: What about a proper variable name, e.g. `linksDown`?

Comment (viktorianer) [reply sernet.gs.service/src/sernet/verinice/model/common/CnATreeElement.java]: It is just a refactored old code. But I am going to change it to.

Comment (viktorianer) [reply sernet.gs.service/src/sernet/verinice/model/common/CnATreeElement.java]: done

File: sernet.gs.service/src/sernet/verinice/service/commands/RemoveElement.java [hunk 2]
```diff
@@ -338,7 +340,7 @@ private boolean hasOrphanEntity(CnATreeElement element) {
     }
 
     private void deleteOrphanEntity(CnATreeElement element) {
-        if (element.getEntity() != null) {
```
Comment (murygin) [inline sernet.gs.service/src/sernet/verinice/service/commands/RemoveElement.java]: Make sure that the old delete operation remains unchanged if the following applies: `module-templates.active=false`

Please remove the next if statement from this command and implement another solution.

**Comment by murygin** [comment]
The two commands should not be changed for the module templates. If the feature toggle is  `module-templates.active=false`, use a different implementation.

**Pull request closed without merging**